vite.base.config.ts

```diff
@@ -0,0 +1,78 @@
+import {UserConfig, defineConfig} from "vite";
+import {nodePolyfills} from "vite-plugin-node-polyfills";
+import {visualizer} from "rollup-plugin-visualizer";
+import topLevelAwait from "vite-plugin-top-level-await";
+import {blsBrowserPlugin} from "./scripts/vite/plugins/blsBrowserPlugin.js";
+
+export function getBaseViteConfig(
+  pkgInfo: {
+    description: string;
+    version: string;
+    author: string;
+    license: string;
+    homepage: string;
+  },
+  {entry, libName}: {entry: string; libName: string}
+): UserConfig {
+  // TODO: Investigate why this banner is not appended to the build header.
+  const banner =
+    `/* ${pkgInfo.description}\n` +
+    " * \n" +
+    ` * Version: ${pkgInfo.version}\n` +
+    ` * Author: ${pkgInfo.author}\n` +
+    ` * License: ${pkgInfo.license}\n` +
+    ` * Web: ${pkgInfo.homepage}\n` +
+    "*/";
+
+  return defineConfig({
+    plugins: [
+      topLevelAwait(),
+      blsBrowserPlugin(),
+      nodePolyfills({
```

Comment:
Shouldn't this handle `node:crypto` too? It's also considered in `blsBrowserPlugin`

Reply:
Yes it can polyfill the whole package, that increases the bundle size with almost 1MB. So why I skipped it from here and replaced with the empty module. 

I tried a lot of tricks to reduce the bundle size, if you run `build:dist` locally under `packages/light-client`, you will notice a file generated `stats.html`, here you can explore what was included in the bundle.